components/org.wso2.carbon.extension.identity.sso.jwt/src/main/java/org/wso2/carbon/identity/sso/jwt/factory/JWTInboundIdentityResponseFactory.java

```diff
@@ -0,0 +1,164 @@
+import org.wso2.carbon.identity.application.authentication.framework.inbound.HttpIdentityResponse;
+import org.wso2.carbon.identity.application.authentication.framework.inbound.HttpIdentityResponseFactory;
+import org.wso2.carbon.identity.application.authentication.framework.inbound.IdentityResponse;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.sso.jwt.message.JWTInboundResponse;
+import org.wso2.carbon.identity.sso.jwt.util.JWTInboundConstants;
+import org.wso2.carbon.identity.sso.jwt.util.JWTInboundUtil;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * This class represents a factory for JWT IdentityResponse instances which will result from the framework,
+ * after the authentication step.
+ *
+ * This class is extensible either by implementing the HttpIdentityResponseFactory or extending the
+ * JWTInboundIdentityResponseFactory. The customized class should be defined under
+ * <JWTSSO><ResponseFactory></ResponseFactory></JWTSSO> in identity.xml. Otherwise JWTInboundIdentityResponseFactory
+ * will be used as the default request processor.
+ */
+public class JWTInboundIdentityResponseFactory extends HttpIdentityResponseFactory {
+
+    private static Log log = LogFactory.getLog(JWTInboundIdentityResponseFactory.class);
+
+    public String getName() {
+
+        return "JWTSSOInboundIdentityResponseFactory";
+    }
+
+    /**
+     * Checks if an incoming IdentityResponse from the framework can be handled by this particular factory.
+     *
+     * @param identityResponse incoming IdentityResponse from the identity framework
+     * @return true if the incoming response is of the type handled by this factory
+     */
+    @Override
+    public boolean canHandle(IdentityResponse identityResponse) {
+
+        return identityResponse instanceof JWTInboundResponse;
+    }
+
+    /**
+     * Converts the received IdentityResponse instance to an HTTPResponse so that it could be sent to the calling party.
+     * This is where the logic for picking up and setting any parameters/headers/cookies etc is written.
+     *
+     * @param identityResponse the received (and handle-able IdentityResponse instance
+     * @return a corresponding HTTPResponse in the form of a builder, so that it could be built on demand
+     */
+    @Override
+    public HttpIdentityResponse.HttpIdentityResponseBuilder create(IdentityResponse identityResponse) {
+
+        HttpIdentityResponse.HttpIdentityResponseBuilder builder
+                = new HttpIdentityResponse.HttpIdentityResponseBuilder();
+
+        if (identityResponse instanceof JWTInboundResponse) {
+            JWTInboundResponse jwtInboundResponse = (JWTInboundResponse) identityResponse;
+
+            String logoutUrl = jwtInboundResponse.getLogoutUrl();
+            if (StringUtils.isBlank(jwtInboundResponse.getToken()) && StringUtils.isNotBlank(logoutUrl)) {
```

Comment:
Shall we break the condition to a private method like `isLogoutRequest` that would improve readability

Reply:
Fixed